src/Microsoft.Health.Fhir.Core/Features/Conformance/ICapabilityStatementBuilder.cs

```diff
@@ -16,17 +15,17 @@ public interface ICapabilityStatementBuilder
 
         ICapabilityStatementBuilder AddRestInteraction(string systemInteraction);
 
-        ICapabilityStatementBuilder AddSearchParams(string resourceType, IEnumerable<SearchParamComponent> searchParameters);
-
         ICapabilityStatementBuilder UpdateRestResourceComponent(string resourceType, Action<ListedResourceComponent> action);
 
         ICapabilityStatementBuilder Update(Action<ListedCapabilityStatement> action);
 
         ICapabilityStatementBuilder AddDefaultResourceInteractions();
 
-        ICapabilityStatementBuilder AddDefaultSearchParameters();
+        ICapabilityStatementBuilder SyncSearchParameters();
+
+        ICapabilityStatementBuilder SyncProfiles();
 
-        ICapabilityStatementBuilder AddDefaultRestSearchParams();
+        ICapabilityStatementBuilder AddSharedSearchParameters();
```

Comment:
Should we rename/distinguish that these are `BuiltIn` or `Embedded` SearchParameters?

Reply:
I'm not sure what does `BuiltIn` or `Embedded` means, but I know what that thing is doing!
Search parameter can be specific for an resource, but there is search parameters which are applicable for all resources, like LastUpdated or Id, or Meta. I call them Shared, I think that's are improvement over `DefaultRest`

If you talking about having difference between something set at server start and never can be changed and things which are mutable through life of server... Probably it need better name. So do some kind of proper refactoring.
At this point I'm just trying to see would this changes work in not my machine setting.

Reply:
Oh ok, then I think "Base" or "Global", but perhaps just a comment then :)

Reply:
I like `Global` more. I also sprinkle few comments around code.

---
In reply to: [606404241](https://github.com/microsoft/fhir-server/pull/1795#discussion_r606404241) [](ancestors = 606404241)